kserve-module/tests/e2e/conftest.py

```
@@ -152,13 +152,19 @@ def _wait_for_managed_deployments_gc(kubectl_bin, is_openshift, timeout=TIMEOUT_
 # ---------------------------------------------------------------------------
 @pytest.fixture(scope="session")
 def cluster_info():
-    """Detect cluster type by checking for OpenShift API resources."""
+    """Detect cluster type and pick the right CLI binary (oc or kubectl)."""
+    import shutil
+
+    cli = "oc" if shutil.which("oc") else "kubectl"
+    if not shutil.which(cli):
+        pytest.fail(f"Neither 'oc' nor 'kubectl' found in PATH")
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)